controller/settingscontroller.php

```diff
@@ -12,20 +12,46 @@
 
 use OCP\IRequest;
 use OCP\AppFramework\Controller;
-use OCA\AgreeDisclaimer\AppInfo\Application;
+use OCP\IAppConfig;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+use OCP\ILogger;
+
 use OCA\AgreeDisclaimer\Utils;
-use \OCP\IL10N;
 
 /**
  * Controller to retreive some application settings through ajax
  */
 class SettingsController extends Controller {
 
+    private $appConfig;
+    private $filePrefix;
+    private $l10n;
+    private $urlGenerator;
+    private $logger;
+    private $loggerParameters;
+    private $utils;
+    private $txtPath;
+    private $pdfPath;
+
     /**
-     * Creates an instance to the SettingsController 
+     * Creates an instance to the SettingsController
+     * @param IAppConfig
      */
-    public function __construct($AppName, IRequest $request) {
+    public function __construct($AppName, IRequest $request,
+                                IAppConfig $appConfig, $filePrefix, IL10N $l10n,
```

Comment:
If a type hint is given, the container will query by full namespace path, if not, it will query by parameter name